src/api/userAPI.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseAPI } from "./baseAPI";
+
+export class UserAPI extends BaseAPI {
+  constructor() {
+    super("/user");
+  }
+
+  updateUserData(data): Promise<unknown> {
+    return this.httpService.put("/profile", {
+      data,
+      headers: { "Content-Type": "application/json" }
+    });
+  }
```

Comment:
в types/api есть тип UserDTO, его можно использовать тут вместо unknown, для этого правда придется немного изменить код httpService, добавить туда type parameters в методы request и put. Код от этого станет гибче, потому что можно будет указывать тип ответа. Я оставил комменты в http.service.ts что и где следует добавить
```suggestion
  updateUserData(data): Promise<UserDTO> {
    return this.httpService.put<UserDTO>("/profile", {
      data,
      headers: { "Content-Type": "application/json" }
    });
  }
```